fili-core/src/test/groovy/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandlerSpec.groovy

```diff
@@ -0,0 +1,545 @@
+        )
+        int expectedCount = 60
+        int limit = 100
+        SuccessCallback classicSuccessCallback = handler.buildSuccessCallback(context, request, groupByQuery, response, limit)
+        SuccessCallback success = handler.buildCacheSuccessCallback(classicSuccessCallback, cacheService, groupByQuery, response)
+        String weightResponse = """
+        [ {
+            "version" : "v1",
+            "timestamp" : "2012-01-01T00:00:00.000Z",
+            "event" : {
+                "count" : "$expectedCount"
+            }
+        } ]
+        """
+        JsonParser parser = new JsonFactory().createParser(weightResponse)
+        JsonNode jsonResult = MAPPER.readTree(parser)
+
+        expect:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 0
+
+        when:
+        success.invoke(jsonResult)
+
+        then:
+        1 * next.handleRequest(context, request, groupByQuery, response)
+
+        and:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 0
+    }
+
+    def "Test build and invoke cache success callback count too high"() {
+        setup:
+        SimplifiedIntervalList intervals = new SimplifiedIntervalList()
+        ResponseContext responseContext =  createResponseContext([(MISSING_INTERVALS_CONTEXT_KEY.name): intervals])
+        response.getResponseContext() >> responseContext
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        CacheWeightCheckRequestHandler handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        )
+        int expectedCount = 200
+        int limit = 100
+        SuccessCallback classicSuccessCallback = handler.buildSuccessCallback(context, request, groupByQuery, response, limit)
+        SuccessCallback success = handler.buildCacheSuccessCallback(classicSuccessCallback, cacheService, groupByQuery, response)
+        String weightResponse = """
+        [ {
+            "version" : "v1",
+            "timestamp" : "2012-01-01T00:00:00.000Z",
+            "event" : {
+                "count" : "$expectedCount"
+            }
+        } ]
+        """
+        JsonParser parser = new JsonFactory().createParser(weightResponse)
+        JsonNode jsonResult = new ObjectMapper().readTree(parser)
+        HttpErrorCallback ec = Mock(HttpErrorCallback)
+
+        expect:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 0
+
+        when:
+        success.invoke(jsonResult)
+
+        then:
+        0 * next.handleRequest(context, request, groupByQuery, response)
+        1 * response.getErrorCallback(groupByQuery) >> ec
+        1 * ec.dispatch(507, _, _)
+
+        and:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 0
+    }
+
+    def "Test build and invoke cache success callback invalid json"() {
```

Comment:
So is this test to test that the success callback works correctly?
Because in the current implementation all it does is invoke the success callback it was passed with possible errors.

Or is this implicitly testing the parent success callback?
If so, is it testing in the spec for that request handler?  If it's not well tested there, feel free to push this test up to test that handler.

Reply:
Some of the tests were to test the parent success callback but we do have tests in the parent handler to check the success callback. I will go ahead and remove them from the CacheWeightCheckRequestHandler test.